src/imperative/imperative_utils.h

```diff
@@ -1002,6 +1009,18 @@ void RunGraph(const bool retain_graph,
               const DispatchModeVector &dispatch_modes,
               bool recording);
 
+
+void NaiveRunGraph(const bool retain_graph,
```

Comment:
I think these new functions deserve documentations for inputs/outputs and what it does to keep the readability